src/main/resources/db/changesets/20240426_addsnapshotidtorequest.yaml

```diff
@@ -0,0 +1,18 @@
+databaseChangeLog:
+  - changeSet:
+      id: add_snapshot_id_to_request
+      author: rjohanek
+      changes:
+        - addColumn:
+            tableName: snapshot_request
+            columns:
+              - column:
+                  name: snapshot_id
```

Comment:
Do we think snapshot_id is clear enough? it contrasts with id (the id of the snapshot request), so that might be clear enough, but we could also call it something like base_snapshot_id if we think that is more easily understood

Reply:
I think this table will need to contain two different snapshot IDs. This one and the ID of the snapshot that's created when the request has been accepted. So I think a more specific name is needed here. Personally I would use `source_snapshot_id` rather than `base_snapshot_id`; we have the concept of "source datasets" for snapshots so this seems like a similar case to me.

Reply:
okay should I add that new column too (for the created snapshot) while I'm here?

Reply:
Yeah, I like `source_snapshot_id` as it matches the naming of other TDR concepts. 
I would encourage sticking the original scope of this ticket (so, not adding it here), but not a big deal either way.

Reply:
Okay, just adding the new source column